### PR TITLE
feat: add useCookieState hook with tests

### PR DESCRIPTION
**Pull request by pikaatic**

### CODE REVIEW

File: src/hooks/useCookieState.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  string | null,
+  (newValue: string, options?: Cookies.CookieAttributes) => void,
+  () => void
+] => {
+  const [value, setValue] = useState<string | null>(
+    () => Cookies.get(cookieName) || null
+  );
+
+  const updateCookie = useCallback(
+    (newValue: string, options?: Cookies.CookieAttributes) => {
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:16]: The `options` here are best passed at `useCookie` level. 

```jsx
const [] = useCookie('my-cookie', options)
```

File: src/hooks/useCookieState.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import Cookies from 'js-cookie';
+import { useCallback, useState } from 'react';
+
+const useCookieState = (
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:4]: JSDoc comments are missing. We need to add them here :D

File: src/hooks/useCookieState.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  cookieName: string
+): [
+  string | null,
+  (newValue: string, options?: Cookies.CookieAttributes) => void,
+  () => void
+] => {
+  const [value, setValue] = useState<string | null>(
+    () => Cookies.get(cookieName) || null
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:12]: `Get` should also use the same options that were used in `set`

Comment (imbhargav5) [reply src/hooks/useCookieState.ts:12]: Als this breaks on the server side. So on the server side we need to return null if window is not defined.

**Review by imbhargav5 — CHANGES_REQUESTED** [review]
Also it would be great if we support this syntax as well. https://github.com/js-cookie/js-cookie#read

**Comment by imbhargav5** [comment]
Looking good overall @iampika Great work!

File: src/hooks/useCookieState.ts (hunk 2 above)
Comment (imbhargav5) [reply src/hooks/useCookieState.ts:15]: So some comments are still missing. You can refer to the other hooks for guidance. Basically we need `useCookie` and also the description of what this hook does in there as well

File: src/hooks/useCookieState.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * @param cookieName Name of the cookie
+ * @returns {handler} A handler to interact with the cookie
+ */
+const useCookieState = (
+  cookieName: string,
+  cookieValue: string | null = null,
+  options?: Cookies.CookieAttributes
+): UseCookieStateType => {
+  const [value, setValue] = useState<string | null>(
+    () => Cookies.get(cookieName, options) || cookieValue
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:21]: We need to check for `window` here. so 

```
const [value, setValue] = useState<string | null>(
     () => {
   if(typeof window !== 'null'){
      return Cookies.get(cookieName, options) || cookieValue
   }else{
      return null;
   }
});
```

File: src/hooks/useCookieState.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    [cookieName, options]
+  );
+
+  const deleteCookie = useCallback(() => {
+    Cookies.remove(cookieName);
+    setValue(null);
+  }, [cookieName]);
+
+  if (!window) return null;
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:37]: We don't need window check here.

**Review by imbhargav5 — CHANGES_REQUESTED** [review]

File: src/hooks/useCookieState.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import { useCallback, useState } from 'react';
+
+type UseCookieStateType = {
+  value: string | null;
+  updateCookie: (newValue: string) => void;
+  deleteCookie: () => void;
+} | null;
+
+/**
+ *
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:11]: We need more comments here.

File: src/hooks/useCookieState.ts [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  options?: Cookies.CookieAttributes
+): UseCookieStateType => {
+  const [value, setValue] = useState<string | null>(() => {
+    if (typeof window !== undefined) {
+      return Cookies.get(cookieName, options) || cookieValue;
+    }
+
+    return null;
+  });
+
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:27]: We need useCookieState to listen for updates to this cookie from external events. So basically a listener on cookie change similar to how our useLocalstorageState works. WE need something for cookies .

**Review by imbhargav5 — CHANGES_REQUESTED** [review]

File: src/hooks/useCookieState.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    [cookieName, options]
+  );
+
+  const deleteCookie = useCallback(() => {
+    Cookies.remove(cookieName);
+    setValue(null);
+  }, [cookieName]);
+
+  return { deleteCookie, updateCookie, value };
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:41]: `[value, setValue, removeValue]`

**Review by imbhargav5 — CHANGES_REQUESTED** [review]

File: src/hooks/useCookieState.ts [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+} | null;
+
+/**
+ *
+ * @param cookieName Name of the cookie
+ * @returns {handler} A handler to interact with the cookie
+ */
+const useCookieState = (
+  cookieName: string,
+  cookieValue: string | null = null,
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:17]: Rename to `key` & 'initialState'.

File: src/hooks/useCookieState.ts [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  options?: Cookies.CookieAttributes
+): UseCookieStateType => {
```
Comment (imbhargav5) [inline src/hooks/useCookieState.ts:19]: 4th argument JSCookie arguments. At least have `converter` option.

**Review by imbhargav5 — CHANGES_REQUESTED** [review]

**Comment by qiweiii** [comment]
@imbhargav5 Seems ready to merge after sovling conflicts

**Pull request closed without merging**